packages/xrpc-server/src/stream/stream.ts

```diff
@@ -11,3 +12,15 @@ export async function* byFrame(ws: WebSocket, options?: DuplexOptions) {
     yield Frame.fromBytes(chunk)
   }
 }
+
+export async function* byMessage(ws: WebSocket, options?: DuplexOptions) {
+  for await (const frame of byFrame(ws, options)) {
+    if (frame.isMessage()) {
+      yield frame
+    } else if (frame.isError()) {
+      throw new XRPCError(-1, frame.code, frame.message)
+    } else {
+      throw new XRPCError(ResponseType.Unknown, undefined, 'Unknown frame type')
```

Comment:
I see that Infos are still around but not handled here, will they get thrown?

Reply:
Good eye.  That's right, they will get thrown.  I figured it didn't make sense to figure out how to fit them into this interface since they are about to disappear, should have left a note about that!  Think it's alright?

Reply:
yup that's fine 👍 

I'd say just leave a note for now. or yield them as that will more closely match the actual behavior

Reply:
Good call, I've started yielding them here 👍